server/weaverbird/steps/totals.py

```diff
@@ -0,0 +1,85 @@
+from typing import List
+
+import pandas as pd
+from pydantic import BaseModel, Field
+
+from weaverbird.steps import AggregateStep, BaseStep
+from weaverbird.steps.aggregate import Aggregation
+from weaverbird.types import ColumnName, DomainRetriever, PipelineExecutor
+
+
+class TotalDimension(BaseModel):
+    total_column: ColumnName
+    total_rows_label: str
+
+
+class TotalsStep(BaseStep):
+    name = Field('totals', const=True)
+    total_dimensions: List[TotalDimension] = Field(alias='totalDimensions')
+    aggregations: List[Aggregation] = Field(min_items=1)
+    groups: List[ColumnName] = Field(min_items=0)
+
+    class Config:
+        allow_population_by_field_name = True
```

Comment:
We could put this into the BaseStep ? (if that propagates, I'm not sure of this ^^)